### PR TITLE
[Fix #10919] Fix a huge performance regression between 1.32.0 and 1.33.0

### LINKED ISSUE #10919: Huge performance regression between 1.32.0 and 1.34.1
## Expected behavior

RuboCop should run in a reasonable amount of time.

## Actual behavior

When we upgraded from RuboCop 1.32.0 to 1.34.1 we saw a huge jump in how long it takes to run a build. What would consistently take < 5 minutes can now take upwards of an hour. I tested on 1.35.0 as well and we are seeing the same issue.

## Steps to reproduce the problem

I unfortunately don't have this, yet. But I'll continue poking around to see if a particular rule is causing the issue. If anybody has any suggestions on places to look that might help.

## RuboCop version

```
$ bundle exec rubocop -V
1.35.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.0.4 x86_64-linux)
  - rubocop-graphql 0.14.5
  - rubocop-performance 1.14.3
  - rubocop-rails 2.15.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.12.1```
```

### PR DESCRIPTION
**Pull request by ydah**
Fix: #10919

Before:
- 1.32.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 2.7.2 x86_64-darwin21)

`bundle exec rubocop --cache false --only   16.83s user 1.30s system 93% cpu 19.320 total`

- 1.35.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 2.7.2 x86_64-darwin21)

`bundle exec rubocop --cache false --only   1255.21s user 5.32s system 71% cpu 29:17.76 total`

After:

`bundle exec rubocop --cache false --only 16.64s user 1.25s system 94% cpu 19.002 total`

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/mixin/allowed_methods.rb [hunk 1]
```diff
@@ -17,16 +17,21 @@ def allowed_method?(name)
 
       # @api public
       def allowed_methods
```
Comment (ydah) [inline lib/rubocop/cop/mixin/allowed_methods.rb:19]: [memo] The reason for the slow speed was due to this method.
If the following settings were used

```yaml
Style/MethodCallWithArgsParentheses:
  IgnoredMethods:
    - abort
    - after_save
    - and
    - before_save
    - delegate
    - describe
    - exec
    - exit
    - load
    - not_to
    - p
    - print
    - puts
    - raise
    - require
    - shared_context
    - shared_examples
    - star_puts
    - throw
    - to
    - warn
    - yield
```

it was returning the following results (increasing with each file)
```
["abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield"]
```

File: spec/rubocop/cli/auto_gen_config_spec.rb [hunk 2]
```diff
@@ -951,33 +951,6 @@ def a; end
       YAML
     end
 
-    context 'when configurable parameter is an obsolete parameter' do
```
Comment (ydah) [inline spec/rubocop/cli/auto_gen_config_spec.rb:954]: This was a test added in <https://github.com/rubocop/rubocop/pull/10875>. However, this specification was incorrect. In fact, I think the cop_config was broken, causing duplicates to be displayed.

Comment (koic) [reply spec/rubocop/cli/auto_gen_config_spec.rb:954]: Isn't it necessary to update the test as originally intended? I'm worried about losing a regression spec for #10875.

Comment (ydah) [reply spec/rubocop/cli/auto_gen_config_spec.rb:954]: Indeed it is. I will modify the test a bit and add it.

**Comment by dvandersluis** [comment]
Good catch! Would any other cops changed in #10829 be affected?

File: spec/rubocop/cli/auto_gen_config_spec.rb (hunk 2 above)
Comment (ydah) [reply spec/rubocop/cli/auto_gen_config_spec.rb:954]: Hmmm. I've tried running the following test and it seems to fail.
Maybe there is some other problem.
Should we include the resolution of that issue as part of this PR?

```ruby
    context 'when duplicated parameter' do
      it 'parameters are displayed without duplication' do
        create_file('.rubocop.yml', <<~YAML)
          AllCops:
            NewCops: enable
          Lint/AmbiguousBlockAssociation:
            AllowedMethods:
              - change
              - not_change
              - change
        YAML
        create_file('example1.rb', ['# frozen_string_literal: true', '', 'def fooBar; end'])
        create_file('example2.rb', ['# frozen_string_literal: true', '', 'def fooBar; end'])

        expect(cli.run(['--auto-gen-config'])).to eq(0)
        File.readlines('.rubocop_todo.yml')
        expect(File.readlines('.rubocop_todo.yml')[9..].join)
          .to eq(<<~YAML)
            # Configuration parameters: AllowedPatterns, IgnoredPatterns.
            # SupportedStyles: snake_case, camelCase
            # AllowedPatterns: change, not_change
            Naming/MethodName:
              EnforcedStyle: camelCase
          YAML
      end
    end
```

```
Failures:

  1) RuboCop::CLI --auto-gen-config --auto-gen-config when duplicated parameter parameters are displayed without duplication
     Failure/Error:
       expect(File.readlines('.rubocop_todo.yml')[9..].join)
         .to eq(<<~YAML)
           # Configuration parameters: AllowedPatterns, IgnoredPatterns.
           # SupportedStyles: snake_case, camelCase
           # AllowedPatterns: change, not_change
           Naming/MethodName:
             EnforcedStyle: camelCase
         YAML
     
       expected: "# Configuration parameters: AllowedPatterns, IgnoredPatterns.\n# SupportedStyles: snake_case, camelCase\n# AllowedPatterns: change, not_change\nNaming/MethodName:\n  EnforcedStyle: camelCase\n"
            got: "# Configuration parameters: AllowedPatterns, IgnoredPatterns.\n# SupportedStyles: snake_case, camelCase\nNaming/MethodName:\n  EnforcedStyle: camelCase\n"
     
       (compared using ==)
     
       Diff:
       @@ -1,6 +1,5 @@
        # Configuration parameters: AllowedPatterns, IgnoredPatterns.
        # SupportedStyles: snake_case, camelCase
       -# AllowedPatterns: change, not_change
        Naming/MethodName:
          EnforcedStyle: camelCase
       
     # ./spec/rubocop/cli/auto_gen_config_spec.rb:970:in `block (4 levels) in <top (required)>'
     # ./spec/support/cli_spec_behavior.rb:26:in `block (2 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:30:in `block (4 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:30:in `chdir'
     # ./lib/rubocop/rspec/shared_contexts.rb:30:in `block (3 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:7:in `block (2 levels) in <top (required)>'

Finished in 7.11 seconds (files took 1.06 seconds to load)
40 examples, 1 failure
```

**Comment by ydah** [comment]
@dvandersluis Yes. I think so. 
the parameters obsoleted in https://github.com/rubocop/rubocop/pull/10555 had the same problem.

File: spec/rubocop/cli/auto_gen_config_spec.rb (hunk 2 above)
Comment (ydah) [reply spec/rubocop/cli/auto_gen_config_spec.rb:954]: I see. I would like to run the same test code on the old version to check the existing behavior first.

Comment (ydah) [reply spec/rubocop/cli/auto_gen_config_spec.rb:954]: The configuration values like `# AllowedNames: at, by` that appear in `.rubocop_todo.yml` were not configuration values in `.rubocop.yml`, but default values in `config/defautl.yml`.
However, what appeared to be the `.rubocop.yml` configuration values in #10875 . This was due to the default values being corrupted by the problem fixed in this PR.
Therefore, it was necessary to duplicate parameter the `config/defautl.yml` here.

File: spec/rubocop/cli/auto_gen_config_spec.rb [hunk 3]
```diff
@@ -951,29 +951,34 @@ def a; end
...
-      it 'displayed as a new parameter setting value without duplication' do
+    context 'when duplicated default configuration parameter' do
+      before do
+        RuboCop::ConfigLoader.default_configuration['Naming/MethodParameterName']
+                             .merge!('AllowedNames' => %w[at by at])
+      end
```
Comment (ydah) [inline spec/rubocop/cli/auto_gen_config_spec.rb:958]: The configuration values like `# AllowedNames: at, by` that appear in `.rubocop_todo.yml` were not configuration values in `.rubocop.yml`, but default values in `config/defautl.yml`.
However, what appeared to be the `.rubocop.yml` configuration values in #10875 . This was due to the default values being corrupted by the problem fixed in this PR.
Therefore, it was necessary to duplicate parameter the `config/defautl.yml` here.

**Comment by koic** [comment]
Thanks!

**Comment by tjchambers** [comment]
Interesting.

**Pull request merged**